sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ThroughputProperties.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Offer;
+import com.azure.cosmos.implementation.OfferAutoscaleSettings;
+
+/**
+ * Represents throughput of the resources in the Azure Cosmos DB service.
+ */
+public class ThroughputProperties {
+    private final Offer offer;
+
+    ThroughputProperties(Offer offer) {
+        this.offer = offer;
+    }
+
+    /**
+     * Create fixed throughput properties.
+     *
+     * @param throughput the throughput
+     * @return the throughput properties
+     */
+    public static ThroughputProperties createFixedThroughput(int throughput) {
+        return new ThroughputProperties(Offer.createFixedOffer(throughput));
+    }
+
+    /**
+     * Create auto scale throughput properties.
+     *
+     * @param maxAutoScaleThroughput the max auto scale throughput
+     * @param autoUpgradethroughputIncrementPercentage the auto upgrade max throughput increment percentage
+     * @return the throughput properties
+     */
+    public static ThroughputProperties createAutoScaledProvisionedThroughput(
+        int maxAutoScaleThroughput,
+        int autoUpgradethroughputIncrementPercentage) {
```

Comment:
Need more baking time for `autoUpgradethroughputIncrementPercentage`. Can it be exlucded from //build scope?

Reply:
Made this method package private.